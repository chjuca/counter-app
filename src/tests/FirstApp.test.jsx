import { getByText, render } from "@testing-library/react"
import FirstApp from "../FirstApp"

describe("Test <FirstApp/>", () =>{
    test('Should match snapshot', () => {

        const title = "Hola, soy Carlos"
        const {container, getByText, getByTestId} = render(<FirstApp title={title}/>)
        expect (container).toMatchSnapshot()
        expect(getByText(title)).toBeTruthy()

        // const h1 = container.querySelector('h1')
        // expect(h1.innerHTML).toBe(title)


        expect(getByTestId("test-title").innerHTML).toContain(title)

    })

    test("Show subtitle send by props", () => {
        
        const title = "Hola, soy Carlos"
        const subtitle = "soy un subtitulo"
        const {getAllByText} = render(
        <FirstApp 
            title={title}
            subTitle={subtitle}
        />
        )

        expect(getAllByText(subtitle).length).toBe(2)
    })
})