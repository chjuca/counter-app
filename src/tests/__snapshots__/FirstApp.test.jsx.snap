// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp/> Should match snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Carlos
  </div>
  <p />
  <p />
  <p />
</div>
`;
